addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -2069,6 +2084,26 @@ const SnippetOptionWidget = Widget.extend({
             }
         }
     },
+    /**
+     * Used to handle attribute or data attribute value change
+     *
+     * @param {string} value
+     * @param {Object} params
+     * @returns {string|undefined}
+     */
+    _selectAttributeHelper(value, params) {
+        if (params.attributeName) {
+            if (params.saveUnit && !params.withUnit) {
+                // Values that come with an unit are saved without unit as
+                // data-attribute unless told otherwise.
+                value = value.split(params.saveUnit).join('');
+            }
+            return value;
+        }
+        if (params.extraClass) {
+            this.$target.toggleClass(params.extraClass, params.defaultValue !== value);
+        }
```

Comment:
I dare you to reach that `if`

Reply:
Oups !
Also I was wondering, shouldn't we let it crash if there is no attributName instead of doing nothing ? To me it looks like the Select(Data)Attribute is useless without attributName and it will be quicker to figure out why our snippet's attribute isn't updating if it crashes with traceback.

Reply:
Yep, initially there was one big method which handle selectClass and attributes and ... but that's not the case anymore so we can let it crash indeed.